sys/kern/vm_map.c

```diff
@@ -517,6 +523,37 @@ vm_map_t *vm_map_clone(vm_map_t *map) {
   return new_map;
 }
 
+static int cow_page_fault(vm_map_t *map, vm_map_entry_t *ent, size_t off,
```

Comment:
I find in-out function arguments like `anonp` confusing in C. Consider passing in-argument like `vm_anon_t *old` and out-argument `vm_anon_t **newp` to the function.